### PR TITLE
Dev 1.4.0 jdbc optimize

### PR DESCRIPTION
**Pull request by peacewong**
### What is the purpose of the change

EngineConn-Core defines the the abstractions and interfaces of the EngineConn core functions.
The Engine Service in Linkis 0.x is refactored, EngineConn will handle the engine connection 
and session management.

### Related issues/PRs

Related issues: #590
Related pr:#591

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala [hunk 1]
```diff
@@ -44,26 +48,41 @@ import java.sql.{
...
 
-class UJESSQLConnection(private[jdbc] val ujesClient: UJESClient, props: Properties)
+class LinkisSQLConnection(private[jdbc] val ujesClient: UJESClient, props: Properties)
     extends Connection
     with Logging {
-  private[jdbc] var creator = "IDE"
+
+  private[jdbc] var creator = "JDBCDriver"
+
+  private[jdbc] var tableau = false
```
Comment (casionone) [inline linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala:60]: rename to  isTableau?

File: linkis-computation-governance/linkis-jdbc-driver/src/main/java/org/apache/linkis/ujes/jdbc/utils/JDBCUtils.java [hunk 2]
```diff
@@ -41,8 +41,6 @@ public static String convertPattern(final String pattern) {
             continue;
           } else if (c == '%') {
             result.append(".*");
-          } else if (c == '_') {
-            result.append('.');
```
Comment (casionone) [inline linkis-computation-governance/linkis-jdbc-driver/src/main/java/org/apache/linkis/ujes/jdbc/utils/JDBCUtils.java:45]: not support ?

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala [hunk 3]
```diff
@@ -157,9 +176,7 @@ class UJESSQLConnection(private[jdbc] val ujesClient: UJESClient, props: Propert
   override def getMetaData: DatabaseMetaData = throwWhenClosed(new UJESSQLDatabaseMetaData(this))
 
   override def close(): Unit = {
-    JavaConversions
-      .asScalaBuffer(runningSQLStatements)
-      .foreach(statement => Utils.tryQuietly(statement.close()))
+    runningSQLStatements.asScala.foreach(statement => Utils.tryQuietly(statement.close()))
     closed = true
```
Comment (casionone) [inline linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala:180]: add runningSQLStatements.clear()?

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala (hunk 1 above)
Comment (peacewong) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala:60]: rename to tableauFlag

File: linkis-computation-governance/linkis-jdbc-driver/src/main/java/org/apache/linkis/ujes/jdbc/utils/JDBCUtils.java (hunk 2 above)
Comment (peacewong) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/java/org/apache/linkis/ujes/jdbc/utils/JDBCUtils.java:45]: yes

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala (hunk 3 above)
Comment (peacewong) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala:181]: > add runningSQLStatements.clear()?

close will invoke clear

**Pull request merged**